TWTValidation/JSON Validator/AST Nodes/TWTJSONSchemaObjectASTNode.m

```diff
@@ -54,4 +55,57 @@ - (NSSet *)validTypes
     return [NSSet setWithObject:TWTJSONSchemaTypeKeywordObject];
 }
 
+
+- (NSArray *)childrenReferenceNodes
+{
+    NSMutableArray *nodes = [[super childrenReferenceNodes] mutableCopy];
+    [nodes addObjectsFromArray:[self childrenReferenceNodesFromNodeArray:self.propertySchemas]];
+    [nodes addObjectsFromArray:[self childrenReferenceNodesFromNodeArray:self.patternPropertySchemas]];
+    [nodes addObjectsFromArray:self.additionalPropertiesNode.childrenReferenceNodes];
+    [nodes addObjectsFromArray:[self childrenReferenceNodesFromNodeArray:self.propertyDependencies]];
+
+    return nodes;
+}
+
+
+- (TWTJSONSchemaASTNode *)nodeForPathComponents:(NSArray *)path
+{
+    TWTJSONSchemaASTNode *node = [super nodeForPathComponents:path];
+    if (node) {
+        return node;
+    }
+
+    NSString *key = path.firstObject;
+    NSArray *remainingPath = [self remainingPathFromPath:path];
+
+    if ([key isEqualToString:TWTJSONSchemaKeywordProperties]) {
+        return [self nodeForPathComponents:remainingPath fromKeyedNodeArray:self.propertySchemas];
+    }
+    if ([key isEqualToString:TWTJSONSchemaKeywordPatternProperties]) {
+        return [self nodeForPathComponents:remainingPath fromKeyedNodeArray:self.patternPropertySchemas];
+    }
+    if ([key isEqualToString:TWTJSONSchemaKeywordAdditionalProperties]) {
+        return [self.additionalPropertiesNode nodeForPathComponents:remainingPath];
+    }
+    if ([key isEqualToString:TWTJSONSchemaKeywordDependencies]) {
+        return [self nodeForPathComponents:remainingPath fromKeyedNodeArray:self.propertyDependencies];
+    }
```

Comment:
I’d do else-ifs here too. I know it’s basically unnecessary, but it reinforces that these are all distinct cases and actually reduces your line count.